MacOS build: --no-use-pep517 error

This error occurs in all recent PRs.
```
macOS: 12.6.5
pip: 23.1
```

```python
+ python3 -m pip install PyInstaller==4.2 --no-use-pep517

Usage:   
  /Users/runner/work/tribler/tribler/build-env/bin/python3 -m pip install [options] <requirement specifier> [package-index-options] ...
  /Users/runner/work/tribler/tribler/build-env/bin/python3 -m pip install [options] -r <requirements file> [package-index-options] ...
  /Users/runner/work/tribler/tribler/build-env/bin/python3 -m pip install [options] [-e] <vcs project url> ...
  /Users/runner/work/tribler/tribler/build-env/bin/python3 -m pip install [options] [-e] <local project path> ...
  /Users/runner/work/tribler/tribler/build-env/bin/python3 -m pip install [options] <archive url/path> ...

--no-use-pep517 error: It is not possible to use --no-use-pep517
without setuptools and wheel installed.
Error: Process completed with exit code 2.
```

https://github.com/Tribler/tribler/pull/7370
https://github.com/Tribler/tribler/pull/7371

--- a/build/mac/makedist_macos.sh
+++ b/build/mac/makedist_macos.sh
@@ -21,7 +21,7 @@ export RESOURCES=build/mac/resources
 
 python3 -m venv build-env
 . ./build-env/bin/activate
-python3 -m pip install --upgrade pip
+python3 -m pip install pip==23.0.1 # pin pip version to avoid "--no-use-pep517" issues with the latest version
 python3 -m pip install PyInstaller==4.2 --no-use-pep517
 python3 -m pip install --upgrade -r requirements-build.txt
 

